### PR TITLE
Added docstring to scatter_with_legend.py [MEP12]

### PR DESCRIPTION
**Pull request by francoiseprovencher**
We added a docstring to the scatter_with_legend.py as part of the PyLadies meetup led by  @discardthree

### CODE REVIEW

**Comment by tacaswell** [comment]
:+1: will merge after CI passes

File: examples/lines_bars_and_markers/scatter_with_legend.py [hunk 1]
```diff
@@ -1,3 +1,11 @@
+'''
+Demo of a scatter plot (scatter) with a legend. 
+
+Also demonstrates how transparency of the markers 
+can be adjusted by giving ``alpha`` a value between 
+0 and 1.
```
Comment (WeatherGod) [inline examples/lines_bars_and_markers/scatter_with_legend.py]: Can we please showcase this paragraph as an *ideal* example of an example docstring? Clear and concise, and highlighted exactly what this example demonstrates!

**Comment by tacaswell** [comment]
We test on pep8 conformance of the source and examples.  It looks like there are some trailing whitespaces left.  This is something you can configure your editor to automatically remove on save.

```
FAIL: matplotlib.tests.test_coding_standards.test_pep8_conformance_examples
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/matplotlib/matplotlib/venv/lib/python3.5/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/home/travis/build/matplotlib/matplotlib/lib/matplotlib/tests/test_coding_standards.py", line 277, in test_pep8_conformance_examples
    expected_bad_files=expected_bad_files)
  File "/home/travis/build/matplotlib/matplotlib/lib/matplotlib/tests/test_coding_standards.py", line 143, in assert_pep8_conformance
    assert_equal(result.total_errors, 0, msg)
AssertionError: 3 != 0 : Found code syntax errors (and warnings):
/home/travis/build/matplotlib/matplotlib/examples/lines_bars_and_markers/scatter_with_legend.py:2:48: W291 trailing whitespace
/home/travis/build/matplotlib/matplotlib/examples/lines_bars_and_markers/scatter_with_legend.py:4:50: W291 trailing whitespace
/home/travis/build/matplotlib/matplotlib/examples/lines_bars_and_markers/scatter_with_legend.py:5:52: W291 trailing whitespace
----------------------------------------------------------------------
```

**Comment by francoiseprovencher** [comment]
@tacaswell I amended the original commit to remove the trailing whitespaces, should be PEP8 compliant now.

**Comment by WeatherGod** [comment]
appveyor failure is unrelated (conda server unavailable)

**Pull request merged**